### ISSUE TITLE
Further development?

### ISSUE THREAD

**Issue by jankap** [Gimly/vscode-matlab#171]
Hi absolutely *love* this extension, it helps me a lot, especially during long rebase/merge/integration tasks in VSCode.

However, I'm afraid there's no development anymore. Last update is from Dec. 2021, @Gimly has not really been online on GitHub and even worse, SNDST00M has completely deleted his/her account :( 

@rlivings39 is there any chance that Mathworks takes care of the development? Many people like VSCode and let's be honest, the merge editors and git supports in Matlab are by far not that advanced compared to VSCode. Users would still need to have a license anyway.

**Comment by toshiakit** [comment]
Hello @jankap and @gohma231, thanks for sharing your thoughts with respect to using MATLAB in VS Code. I work at MathWorks in community relations role. 

Projects/extensions like these are important for the MATLAB community and we recognize that being able to work with MATLAB code more effectively in VS Code is an important use case. We are very interested in doing more in this space so any feedback is definitely appreciated! 

Stay tuned! 😉

**Comment by jankap** [comment]
> Hello @jankap and @gohma231, thanks for sharing your thoughts with respect to using MATLAB in VS Code. I work at MathWorks in community relations role. 
> 
> Projects/extensions like these are important for the MATLAB community and we recognize that being able to work with MATLAB code more effectively in VS Code is an important use case. We are very interested in doing more in this space so any feedback is definitely appreciated! 
> 
> Stay tuned! 😉
> 

That are awesome news, thank you. Looking forward to seeing some news :)

**Comment by Gimly** [comment]
Hi,

As you might have guessed from my lack of work on the extension, life happened and I got took on some other projects which means that I don't really have time anymore to work on this project.

Even though I had a couple discussions with people at Mathworks over the years, unfortunately nothing much came out of them in terms of helping develop the extension. I don't even have a Matlab license anymore as I'm not using it much today.

All that to say that even though I'm a bit disappointed in not being able to help much anymore and to hear that Mathworks seems to be working on something else, this is life of open source software and I feel that for the community that uses it, it's probably better if someone could take it over.

I would like to discuss how we could do the transition in a way that ensures that the community can continue to contribute and benefit from it in a free and open manner. 

Xavier

**Comment by jankap** [comment]
Dear @Gimly 
life always happens, thank you so much for all your work. I really hope that your efforts here will be honored and probably picked up by Mathworks? I did not understand that Mathworks is working on something very different. @toshiakit could you clarify by any chance? Mathworks could become maintainer of this repo to continue the awesome work of Gimly/Xavier.

**Comment by toshiakit** [comment]
Thanks @Gimly and @jankap for your comments, and I would also like to thank @Gimly for the work he did for the community. I am not familiar with the details, but I sure hope the transition go as smoothly as possible with your help and guidance. Let me check with the team and I will find a way to reach out to you for the next step.

**Comment by toshiakit** [comment]
@Gimly I couldn't find your contact info on your profile. Can you send me an email via my MATLAB Central profile (use "Contact" widget)?
https://www.mathworks.com/matlabcentral/profile/authors/951521/

**Comment by toshiakit** [comment]
@Gimly as @jankap said the best way to honor your efforts is, in my opinion, by trying to do the best we can to serve your users with as smooth a transition as possible, and we are interested to hear your ideas. So please do contact me soon, since we are approaching the holiday season here in the US and our communication may not be as timely as we would like. If you prefer a different way to reach out to me, let me, please let me know.

**Comment by Gimly** [comment]
@toshiakit I've sent you a message through the link you proposed. Looking forward to talking with you.

**Comment by toshiakit** [comment]
Thanks @Gimly , I provided my email address in my response. Please check your spam filter if you haven't seen it yet.

**Comment by meche-gh** [comment]
Is there any possibility of a robust LSP solution? Supporting method calls like `img = readimage(imds,2)`

**Comment by gohma231** [comment]
Any update on this?

**Comment by toshiakit** [comment]
Hi @gohma231, @Gimly and I are exchanging emails now. I am inviting him to meet our dev team in March.

**Comment by zm-cttae** [comment]
The textmate LSP package is now back up on [<s>Gitlab</s>](https://gitlab.com/SNDST00M/vscode-textmate-languageservice) [GitHub](https://github.com/vscode-toolroom/vscode-textmate-languageservice).

**Comment by jankap** [comment]
How cool is that, the author moved to Gitlab and there's active development. Very nice, I gave up hope on that. 👍

**Comment by M00TSDNS** [comment]
You can be sure they're the author because they had the account there since May 26, 2021. 😉

The MATLAB team does indeed have an LSP project with an internal Gitlab repo, however it is entirely unclear how ready that actually is for release. I'm presuming it's not and fixing the Textmate shim package is of value.

**Comment by M00TSDNS** [comment]
@Gimly for now I opened a pull in #174 that I've fully tested on my local machine.. please merge! This will get the issue tracker back in a good place

**Comment by apjanke** [comment]
> The MATLAB team does indeed have an LSP project with an internal Gitlab repo, however it is entirely unclear how ready that actually is for release. I'm presuming it's not for now and fixing the Textmate shim package is of value.

FWIW, I think having a community-supported LSP Matlab project is still of value even if MathWorks makes a public release of their LSP project. Diversity is strength, community projects can experiment more and faster, never trust a commercial software vendor, yadda yadda.

**Comment by M00TSDNS** [comment]
I'm glad that Atom, the inspiration for the syntax grammar shim, also was forked by the community. Symbols were built into Atom's MATLAB experience but it just wasn't configurable.

**Comment by M00TSDNS** [comment]
@Gimly Changelog for this release (`3.0.0`):
- Browser support for language features.
- Update vscode-textmate-languageservice to 1.1.0.
- Large optimization for language features.

Hopefully this can go out soon™.

**Comment by Gimly** [comment]
@M00TSDNS published version 3.0.0, thanks a lot for the awesome work.

**Comment by Gimly** [comment]
Hi everyone, I wanted to inform you that the Mathworks team has released their official Matlab extension, which can be found at https://blogs.mathworks.com/matlab/2023/04/26/do-you-use-visual-studio-code-matlab-is-now-there-too/. As a result, I have made an announcement in the readme that I will no longer maintain this version of the extension. I would like to express my sincere appreciation to all the contributors who have worked on this extension over the years. Your contributions have been instrumental in making it a popular tool for the community. I'm confident that the Mathworks team is eager to continue receiving your great contributions to their own extension.

**Comment by apjanke** [comment]
And we appreciate *you* for your work on this extension. It has been a useful tool in itself, and probably put some pressure on MathWorks to release this official extension. Thank you! Enjoy your rest away from the project.

**Comment by toshiakit** [comment]
Yes as @apjanke said, the success of this repo opened the eyes at MathWorks and without his effort this was not possible. Thank you Xavier! Also, he encouraged MathWorks to accept contributions as he was also helped by his. I am glad our dev team embrace his advice.

**Comment by zm-cttae** [comment]
I have adopted & moved the library powering Gimly's ext to GitHub 🤞🏽
This means I've been given the email in the old package.json also

https://github.com/vscode-toolroom/vscode-textmate-languageservice

I'm welcoming stars, issues and contribution as before! Otherwise, I expect to do work on the official ext also.
